### PR TITLE
ci: add auto review assignment and PR template

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

add auto PR reviewer assignment ci and add a simplified PR template. 

## Summary by CodeRabbit

- **New Features**
	- Introduced automatic code review assignment configuration settings.
	- Added options for default reviewers, reviewer groups, file-specific reviewers, and review settings.

### CODE REVIEW

**Review by jcstein — APPROVED** [review]

File: .github/auto_request_review.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "**":
+    - rollkit
+  ".github/**":
+    - MSevey
+    - rollkit
+options:
+  ignore_draft: true
+  ignored_keywords:
+    - WIP
+  number_of_reviewers: 3
```
Comment (jcstein) [inline .github/auto_request_review.yml:23]: does this randomly rotate reviewers @MSevey?

Comment (MSevey) [reply .github/auto_request_review.yml:23]: yes, it will randomly select 3 from the list. 
If you want to always be assigned I can update it to be that.

Comment (jcstein) [reply .github/auto_request_review.yml:23]: I don't mind being assigned to all, with others being rotated. if it gets to a point where I hit review fatigue, I'll open PR to change it?

Comment (MSevey) [reply .github/auto_request_review.yml:22]: sounds good. I think the following updates should make it so you are always assigned.
If we see PRs where that is not the case we can look at other options.

**Pull request merged**